homeassistant/components/ring/__init__.py

```diff
@@ -50,4 +71,28 @@ def setup(hass, config):
             title=NOTIFICATION_TITLE,
             notification_id=NOTIFICATION_ID)
         return False
+
+    @callback
```

Comment:
I'm assuming `hub_refresh` is doing I/O so this function should not be decorated with `@callback`, ie it should not run in the event loop.

Reply:
ah, thats what that means. fixed.